gcloud-java-dns/src/main/java/com/google/gcloud/dns/testing/LocalDnsHelper.java

```diff
@@ -0,0 +1,1360 @@
+          // there is a match, pass the handling accordingly
+          Response response = pickHandler(exchange, regex);
+          writeResponse(exchange, response);
+          return; // only one match is possible
+        }
+      }
+      // could not be matched, the service returns 404 page not found here
+      writeResponse(exchange, Error.NOT_FOUND.response(String.format(
+          "The url %s does not match any API call.", exchange.getRequestURI())));
+    }
+
+    private Response handleZoneDelete(String projectId, String zoneName) {
+      return deleteZone(projectId, zoneName);
+    }
+
+    private Response handleZoneGet(String projectId, String zoneName, String query) {
+      String[] fields = OptionParsers.parseGetOptions(query);
+      return getZone(projectId, zoneName, fields);
+    }
+
+    private Response handleZoneList(String projectId, String query) {
+      Map<String, Object> options = OptionParsers.parseListZonesOptions(query);
+      return listZones(projectId, options);
+    }
+
+    private Response handleProjectGet(String projectId, String query) {
+      String[] fields = OptionParsers.parseGetOptions(query);
+      return getProject(projectId, fields);
+    }
+
+    /**
+     * @throws IOException if the request cannot be parsed.
+     */
+    private Response handleChangeCreate(HttpExchange exchange, String projectId, String zoneName,
+        String query) throws IOException {
+      String[] fields = OptionParsers.parseGetOptions(query);
+      String requestBody = decodeContent(exchange.getRequestHeaders(), exchange.getRequestBody());
+      Change change = jsonFactory.fromString(requestBody, Change.class);
+      return createChange(projectId, zoneName, change, fields);
+    }
+
+    private Response handleChangeGet(String projectId, String zoneName,
+        String changeId, String query) {
+      String[] fields = OptionParsers.parseGetOptions(query);
+      return getChange(projectId, zoneName, changeId, fields);
+    }
+
+    private Response handleChangeList(String projectId, String zoneName, String query) {
+      Map<String, Object> options = OptionParsers.parseListChangesOptions(query);
+      return listChanges(projectId, zoneName, options);
+    }
+
+    private Response handleDnsRecordList(String projectId, String zoneName, String query) {
+      Map<String, Object> options = OptionParsers.parseListDnsRecordsOptions(query);
+      return listDnsRecords(projectId, zoneName, options);
+    }
+
+    /**
+     * @throws IOException if the request cannot be parsed.
+     */
+    private Response handleZoneCreate(HttpExchange exchange, String projectId, String query)
+        throws IOException {
+      String[] options = OptionParsers.parseGetOptions(query);
+      String requestBody = decodeContent(exchange.getRequestHeaders(), exchange.getRequestBody());
+      ManagedZone zone;
+      try {
+        // IllegalArgumentException if the request body is an empty string
```

Comment:
This exception is being caught, so there is nothing to document. I removed the comment completely.